x/nexus/keeper/keeper.go

```diff
@@ -159,6 +159,23 @@ func (k Keeper) EnqueueForTransfer(ctx sdk.Context, sender exported.CrossChainAd
 	k.Logger(ctx).Info(fmt.Sprintf("Transfer of %s to cross chain address %s in %s successfully prepared",
 		asset.Amount.String(), recipient.Address, recipient.Chain.Name))
 
+	// merging all transfers for the specified recipient
+	store := k.getStore(ctx)
+	transfers := k.getPendingTransfersForRecipient(ctx, recipient)
+	merged := mergeTransfersBy(transfers, func(transfer exported.CrossChainTransfer) string {
+		return fmt.Sprintf("%s-%s", transfer.Recipient.Address, transfer.Asset.Denom)
+	})
+
+	for _, transfer := range transfers {
+		key := utils.LowerCaseKey(transfer.Recipient.Chain.Name).
+			Append(utils.LowerCaseKey(strconv.FormatUint(transfer.ID, 10)))
+		store.Delete(utils.LowerCaseKey(exported.Pending.String()).Append(key))
+	}
+
+	for _, transfer := range merged {
+		k.setPendingTransfer(ctx, transfer.Recipient, transfer.Asset)
+	}
```

Comment:
This looks quite weird that it's trying to even touch transfers to the same recipient but of a different asset. It should only attempt to fine the one and only one in the queue to the same recipient and of the same asset.

Reply:
done